### DISCUSSION TITLE
Bootstrap Flux without git?

### DISCUSSION THREAD

**Discussion by stevefan1999-personal** [fluxcd/flux2#4188] [Q&A]
I just want to use the Helm controller CRDs but when I'm using my own Gitea instance installed inside the K8S cluster using Helm. But I also want to manage the Gitea server as a HelmRelease. Well, I can leave it as is and manually maintain the Gitea helm chart myself, but I'm curious about the question what if someone don't want to use Git as a source of bootstrap?

**Comment by kingdonb** [comment]
You can use Flux without a git bootstrap, it will not be GitOps but this is a valid configuration.

You will still need the Source controller running alongside Helm controller because the Source controller manages Helm sources. You will basically just run `flux install` and you can adjust the configuration using `--components` which you can set to only source-controller and helm-controller, I'd recommend keeping the notification controller as well since you might want Alerts and Providers.

There is an alternative Flux without git that is still GitOps, demonstrated here: https://github.com/stefanprodan/flux-aio it uses OCI and Cue instead of Git and it might be something that would interest you.

If you have no interest in running "GitOps mode" you can create OCIRepository, HelmRepository, resources manually on the cluster with kubectl apply or `flux create` and you can exercise them with a HelmRelease. Then your cluster's source of truth is just etcd.
